packages/realm/src/app-services/App.ts

```diff
@@ -187,24 +189,24 @@ export class App {
     return new EmailPasswordAuth(internal);
   }
 
-  public get currentUser(): User | null {
+  public get currentUser(): User<FunctionsFactoryType, CustomDataType> | null {
```

Comment:
Just a thought: Instead of typing out this user type multiple times, it could make sense to add another type argument to `App` based on the other arguments, to make a local `AppUser` type alias.

I'm thinking something like: https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEkoGdnwKrJDAPAFQD54BvAKHgvgAcYB7AMwEsIQAuePAblIF9TTQSOImip4AQSpV8AGglTM2eAF4MWXISJlK8AK7r2kqopjc+QA

Again - just a thought.

Reply:
I do think that makes sense, but is that a breaking change?  I would have to play it out.  Perhaps my assumptions are wrong or I'm not completely getting the picture

Reply:
Why are we only supplying 2 of the 3 parameters to `User<...>`? Maybe `App` should be templated on `UserType extends AnyUser = User` instead those details and just return `UserType` here?

Reply:
> Maybe `App` should be templated on `UserType`

(How) is that different from what I suggested above?

Reply:
If I understand correctly, you are proposing to add an additional parameter. I am instead suggesting to replace all of the parameters with a single `UserType` parameter.

Reply:
Okay, yeah - that would be a breaking change though, which we should probably postpone to v13?

We're tracking a proposal to collapse the type signature on `App` which feels related to me: https://github.com/realm/realm-js/issues/3593